openedx_learning/apps/authoring/publishing/api.py

```diff
@@ -644,87 +782,407 @@ def _add_to_existing_draft_change_log(
+    # Subsection's side-effect on its Section each time through the loop.
+    # It also guards against infinite parent-child relationship loops, though
+    # those aren't *supposed* to happen anyhow.
     processed_entity_ids: set[int] = set()
-    for change in change_log.records.all():
-        _create_container_side_effects_for_draft_change(
-            change,
-            processed_entity_ids=processed_entity_ids,
+    for original_change in change_log.records.all():
+        affected_by_original_change = branch_cls.objects.filter(
+            version__dependencies=original_change.entity
         )
+        changes_and_affected = [
+            (original_change, current) for current in affected_by_original_change
+        ]
 
+        # These are the Published or Draft objects where we need to repoint the
+        # log_record (publish_log_record or draft_change_log) to point to the
+        # side-effect changes, e.g. the DraftChangeLogRecord that says, "This
+        # Unit's version stayed the same, but its dependency hash changed
+        # because a child Component's draft version was changed." We gather them
+        # all up in a list so we can do a bulk_update on them.
+        branch_objs_to_update_with_side_effects = []
+
+        while changes_and_affected:
+            original_change, affected = changes_and_affected.pop()
+
+            # If the Draft or Published that is affected by this change is not
+            # already in the change_log, then we have to add it.
+            affected_version_pk = affected.version_id
+
+            change_log_param = {}
+            if branch_cls == Draft:
+                change_log_param['draft_change_log'] = original_change.draft_change_log  # type: ignore[union-attr]
+            elif branch_cls == Published:
+                change_log_param['publish_log'] = original_change.publish_log  # type: ignore[union-attr]
+
+            # Example: If the original_change is a DraftChangeLogRecord that
+            # represents editing a Component, the side_effect_change is the
+            # DraftChangeLogRecord that represents the fact that the containing
+            # Unit was also altered (even if the Unit version doesn't change).
+            side_effect_change, _created = change_record_cls.objects.get_or_create(
+                **change_log_param,
+                entity_id=affected.entity_id,
+                defaults={
+                    # If a change record already exists because the affected
+                    # entity was separately modified, then we don't touch the
+                    # old/new version entries. But if we're creating this change
+                    # record as a pure side-effect, then we use the (old_version
+                    # == new_version) convention to indicate that.
+                    'old_version_id': affected_version_pk,
+                    'new_version_id': affected_version_pk
```

Comment:
```suggestion
                    'old_version_id': affected.version_id,
                    'new_version_id': affected.version_id,
```
(this goes with my previous suggestion)